jdaviz/configs/mosviz/plugins/parsers.py

```diff
@@ -309,13 +316,9 @@ def _parse_as_spectrum1d(path):
 
 
 def _load_fits_image_from_filename(filename, app):
-    data_list = []
     with fits.open(filename) as hdulist:
-        meta = dict(hdulist[0].header.copy())
-        data_iter = get_image_data_iterator(app, hdulist, "Image", ext=None)
-        for d, _ in data_iter:  # We do not use the generated labels
-            d.meta.update(meta)
-            data_list.append(d)
+        # We do not use the generated labels
+        data_list = [d for d, _ in get_image_data_iterator(app, hdulist, "Image", ext=None)]
```

Comment:
Metadata stuff is already taken care of in `get_image_data_iterator` (Imviz parsing logic).